pkg/resolvers/grbClusterResolver.go

```diff
@@ -86,6 +109,74 @@ func (g *GRBClusterRuleResolver) VisitRulesFor(user user.Info, namespace string,
 			return
 		}
 	}
+	// if we aren't an exception service account, no need to check add sa-specific permissions
+	if !isExceptionServiceAccount(user) {
```

Comment:
Would it be less risky to start with this check?

Let me know how wrong I got this but here's what I think the requirement is:
- No rules should be returned for any serviceaccounts except for "exceptional" SA which passe a SAR `*` check.

So then, how risky would it be if any of the code above accumulate rules for a service account? Right now we assume that `GetByIndex` for groups and user won't return anything for a serviceaccount (and the tests below also hardcode this assumption). Is this always the case? Also, what if that code above is modified later to visit some rules and by mistake this leaks for SA too.

IOW, would it be safer / more future proof to do this:

```go
func (g *GRBClusterRuleResolver) VisitRulesFor(user user.Info, namespace string, visitor func(source fmt.Stringer, rule *rbacv1.PolicyRule, err error) bool) {
    if isAServiceAccount() {
       if isExceptionServiceAccount() {
        // handle exception
        }
     // Make sure no rules for SA leaks
     return
    }
    // Regular users
```

Does that make sense?